### ISSUE TITLE
Headings should have content

### ISSUE THREAD

**Issue by alvaromontoro** [jackdomleo7/Checka11y.css#33]
### Describe the new a11y feature or project enhancement

Headings are essential to communicate the organization and flow of a page. An empty/unreadable heading does not make sense from a semantic point of view, it could create problems with assistive technologies, and is a sign of an underlying problem (e.g., using tags/elements for visual results instead of for what they are supposed to mean). 

And quoting [Deque's site](https://dequeuniversity.com/rules/axe/4.0/empty-heading): "_It is a best practice to ensure all heading elements (those marked with `<h1>` through `<h6>`) contain content_."

### Describe the solution you'd like

There should be some tests for headings to ensure that they contain content and it's "valid" content. Some of those checks would be making sure that the headings:

- ...are not empty (e.g., using the `:empty` pseudo-class):

      h1:empty,
      h2:empty,
      ... { /* error */ }

- ...are reachable by assistive technology (e.g., they don't have `aria-hidden="true"`)

      h1[aria-hidden="true"],
      h2[aria-hidden="true"],
      ... { /* error */ }

- ...have valid content (e.g., the content is not a single image element)

      h1 img:only-child { /* error */ }

### Link(s)

- [MDN: `:empty` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:empty)
- [MDN: `:only-child` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:only-child)
- [Deque University: empty headings rule](https://dequeuniversity.com/rules/axe/4.0/empty-heading)
- [WebAIM: Headings definition and implementation](https://webaim.org/techniques/semanticstructure/#headings)

**Comment by jackdomleo7** [comment]
I like this. I think we can further improve this to look for any empty element that shouln't be?

**Comment by danspratling** [comment]
Aren't some elements valid without content?

One that springs to mind is `<i>` which is often used for icons `<i class=''fas fa-wheelbarrow"></i>`

**Comment by alvaromontoro** [comment]
Can I pick it this issue up?
